app/controllers/api/slots_controller.rb

```diff
@@ -1,8 +1,15 @@
 module Api
   class SlotsController < ApiController
     def index
-      prison = Prison.enabled.find(params[:prison_id])
-      @slots = prison.available_slots
+      prison = Prison.enabled.find(params.fetch(:prison_id))
+
+      slot_availability = SlotAvailability.new(prison: prison)
+      slot_availability.restrict_by_prisoner(
+        prisoner_number: params.fetch(:prisoner_number),
+        prisoner_dob: Date.parse(params.fetch(:prisoner_dob))
+      )
```

Comment:
The reason for this design is that we'll have more restrictions to apply as time continues. Adding all the arguments to the initialiser would make the code less flexible and increasingly convoluted.

A potential alternative design I discussed with @alan would be to make this method a functional filter, returning self with a filtered list of dates, rather than mutating slots. We decided that we'd revisit when we implement the next restriction.